salt/_runners/metalk8s_saltutil.py

```diff
@@ -21,7 +21,12 @@ def sync_auth(saltenv='base', extmod_whitelist=None, extmod_blacklist=None):
 def wait_minions(tgt='*', retry=10):
     client = salt.client.get_local_client(__opts__['conf_file'])
 
-    minions = client.cmd(tgt, 'test.ping', timeout=2)
+    try:
+        minions = client.cmd(tgt, 'test.ping', timeout=2)
+    except Exception as exc:
+        log.warning('Unable to run "test.ping" on "%s": "%s"', tgt, exc)
+        minions = None
+
```

Comment:
Make this a helper function (in the local `wait_minions` scope) and re-use it where needed, instead of copy-pasting (that's too error-prone).

Reply:
This function is already mostly copy/paste but ... ok just rewrited part of the function